packages/number-input/src/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,191 @@
+    const prefix = 'lg-numberinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const errorMessageId = useIdAllocator({ prefix, id: ariaDescribedbyProp });
+    const descriptionId = useIdAllocator({ prefix });
+    const selectId = useIdAllocator({ prefix });
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    const hasUnit = !!unit;
+    const hasSelectOptions =
+      Array.isArray(unitOptions) && unitOptions.length > 1;
+
+    const renderUnitOnly = hasUnit && !hasSelectOptions;
+    const renderSelectOnly = hasUnit && hasSelectOptions;
+    const renderErrorMessage = state === State.Error && errorMessage;
+
+    const popoverProps = {
+      popoverZIndex,
+      usePortal,
+      portalClassName,
+      portalContainer,
+      scrollContainer,
+    } as const;
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <div ref={forwardedRef} className={className}>
+          {(label || description) && (
+            <div className={labelDescriptionStyles}>
+              {label && (
+                <Label htmlFor={inputId} disabled={disabled}>
+                  {label}
+                </Label>
+              )}
+              {description && (
+                <Description id={descriptionId} disabled={disabled}>
+                  {description}
+                </Description>
+              )}
+            </div>
+          )}
+          <div
+            className={cx(wrapperBaseStyles, wrapperSizeStyles[size], {
+              [wrapperGapStyles]: renderUnitOnly,
+            })}
+          >
+            <Input
+              value={value}
+              onChange={onChange}
+              disabled={disabled}
+              size={size}
+              id={inputId}
+              hasSelectOptions={renderSelectOnly}
+              state={state}
+              errorMessage={errorMessage}
+              aria-describedby={`${errorMessageId} ${
+                description ? descriptionId : ''
+              } ${renderSelectOnly ? selectId : ''}`}
+              aria-labelledby={
+                !label && ariaLabelledbyProp ? ariaLabelledbyProp : ''
+              }
+              aria-label={!label && ariaLabelProp ? ariaLabelProp : ''}
+              {...rest}
+            />
+            {renderUnitOnly && (
+              <Overline
+                className={cx(unitBaseStyles, unitThemeStyles[theme], {
+                  [unitDisabledStyles[theme]]: disabled,
+                })}
+              >
+                {unit}
+              </Overline>
+            )}
+            {renderSelectOnly && (
+              <Select
+                id={selectId}
+                disabled={disabled}
+                unit={unit as string}
+                unitOptions={unitOptions}
+                onChange={onSelectChange}
+                size={size}
+                {...popoverProps}
+              />
+            )}
+          </div>
+          <div
+            className={errorMessageStyles}
```

Comment:
are we consistent with these aria-roles? can we also add to style guide please

Reply:
I started adding these roles in the password input. PIn other input components, we were only rendering the error message container if there was an error which is not ideal for screen readers since that message would not be announced when shown. What I'm doing here is always rendering the error message container and adding `aria-live` and `aria-relevant` as attributes. This way whenever the text changes in this container screen readers will announce it. This is something we should incorporate in our form control component.